compaction/compaction_descriptor.hh

```diff
@@ -156,6 +156,12 @@ struct compaction_descriptor {
 
     ::io_priority_class io_priority = default_priority_class();
 
+    class dummy_tag {};
+    using compact_fully_expired = seastar::bool_class<dummy_tag>;
+
+    // Denotes if this compaction task is comprised solely of completely expired SSTables
+    compact_fully_expired is_compact_fully_expired = compact_fully_expired::no;
```

Comment:
let's move `"using compact_fully_expired = seastar::bool_class<dummy_tag>; "` to outside compaction_descriptor, then is_compact_fully_expired can be renamed to compact_fully_expired_sstable, e.g.:

`sstables::compact_fully_expired compact_fully_expired = sstables::compact_fully_expired::no;`

is_compact_fully_expired sounds awkward to me.

Reply:
I agree but I think that even `compact_fully_expired` is not the clearest self-explanatory expression. "Compact fully expired" is confusing because it sounds like, if set to "no", we would not want this compaction to compact fully expired sstables. 

We need some name that would make it clear that this boolean is not about "what to do" but rather "what this task is has to deal with".

Maybe `has_only_fully_expired`?

Reply:
has_only_fully_expired sounds good to me.